wd_helper.au3

```diff
@@ -775,38 +775,40 @@ EndFunc   ;==>_WD_LoadWait
 ;                  - $_WD_ERROR_InvalidDataType
 ;                  - $_WD_ERROR_InvalidExpression
 ; Author ........: Danp2
-; Modified ......:
+; Modified ......: mLipok
 ; Remarks .......:
 ; Related .......: _WD_Window, _WD_ElementAction
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
 Func _WD_Screenshot($sSession, $sElement = Default, $nOutputType = Default)
 	Local Const $sFuncName = "_WD_Screenshot"
-	Local $sResponse, $sResult, $iErr
+	Local $sResponse, $sResult = "", $iErr
 
 	If $sElement = Default Then $sElement = ""
 	If $nOutputType = Default Then $nOutputType = 1
 
 	If $sElement = '' Then
 		$sResponse = _WD_Window($sSession, 'Screenshot')
-		$iErr = @error
 	Else
 		$sResponse = _WD_ElementAction($sSession, $sElement, 'Screenshot')
-		$iErr = @error
 	EndIf
+	$iErr = @error
 
 	If $iErr = $_WD_ERROR_Success Then
+		Local $dBinary
 		Switch $nOutputType
+			Case 1, 2 ; String or Binary - pre processing
+				$dBinary = __WD_Base64Decode($sResponse)
+				If Not @error Then ContinueCase
+				$iErr = $_WD_ERROR_GeneralError
 			Case 1 ; String
-				$sResult = BinaryToString(__WD_Base64Decode($sResponse))
+				$sResult = BinaryToString($dBinary)
 			Case 2 ; Binary
-				$sResult = __WD_Base64Decode($sResponse)
+				$sResult = $dBinary
```

Comment:
@mlipok Have you tested this? I don't think ContinueCase works like this. 😦

Reply:
Yes it works like that I even add new `Switch[2].au3` example file to HelpFile.

Check it with `Switch[2].au3`  code:
```AutoIt
#include <MsgBoxConstants.au3>

_Example()

Func _Example()
	Local $sMsg = ""

	Switch 2
		Case 1
			$sMsg &= " Case 1" & @CRLF
			ContinueCase
		Case 2
			$sMsg &= " Case 2" & @CRLF
			ContinueCase
		Case 1, 2
			$sMsg &= " Case 4" & @CRLF
			ContinueCase
		Case 2 To 3
			$sMsg &= " Case 5" & @CRLF
			ContinueCase ; take care when you use `ContinueCase` just before `Case Else`
		Case Else
			$sMsg &= " Case 6" & @CRLF
	EndSwitch

	MsgBox($MB_SYSTEMMODAL, "", $sMsg)

EndFunc   ;==>_Example

```

Reply:
Oh... you ask if I check `_WD_Screenshot` is still working.
Let see...

Reply:
Hmmm... This was my test:
```autoit
$nOutputType = 2

	Switch $nOutputType
			Case 1, 2 ; String or Binary - pre processing
		ConsoleWrite('@@ Debug(Case 1,2) : $nOutputType = ' & $nOutputType & @CRLF & '>Error code: ' & @error & @CRLF) ;### Debug Console
		ContinueCase
			Case 1 ; String
		ConsoleWrite('@@ Debug(Case 1) : $nOutputType = ' & $nOutputType & @CRLF & '>Error code: ' & @error & @CRLF) ;### Debug Console
			Case 2 ; Binary
		ConsoleWrite('@@ Debug(Case 2) : $nOutputType = ' & $nOutputType & @CRLF & '>Error code: ' & @error & @CRLF) ;### Debug Console
			Case 3 ; Base64
		ConsoleWrite('@@ Debug(Case 3) : $nOutputType = ' & $nOutputType & @CRLF & '>Error code: ' & @error & @CRLF) ;### Debug Console
	EndSwitch
```

This is the resulting output:
```autoit
>"C:\Program Files (x86)\AutoIt3\SciTE\..\AutoIt3.exe" "C:\Program Files (x86)\AutoIt3\SciTE\AutoIt3Wrapper\AutoIt3Wrapper.au3" /run /prod /ErrorStdOut /in "C:\Users\danpo\Dropbox\webdriver\testxxx.au3" /UserParams    
+>13:10:58 Starting AutoIt3Wrapper (21.316.1639.1) from:SciTE.exe (4.4.6.0)  Keyboard:00000409  OS:WIN_10/2009  CPU:X64 OS:X64  Environment(Language:0409)  CodePage:0  utf8.auto.check:4
+>         SciTEDir => C:\Program Files (x86)\AutoIt3\SciTE   UserDir => C:\Users\danpo\AppData\Local\AutoIt v3\SciTE\AutoIt3Wrapper   SCITE_USERHOME => C:\Users\danpo\AppData\Local\AutoIt v3\SciTE 
>Running AU3Check (3.3.14.5)  from:C:\Program Files (x86)\AutoIt3  input:C:\Users\danpo\Dropbox\webdriver\testxxx.au3
+>13:10:58 AU3Check ended.rc:0
>Running:(3.3.14.5):C:\Program Files (x86)\AutoIt3\autoit3.exe "C:\Users\danpo\Dropbox\webdriver\testxxx.au3"    
+>Setting Hotkeys...--> Press Ctrl+Alt+Break to Restart or Ctrl+BREAK to Stop.
@@ Debug(Case 1,2) : $nOutputType = 2
>Error code: 0
@@ Debug(Case 1) : $nOutputType = 2
>Error code: 0
+>13:10:58 AutoIt3.exe ended.rc:0
+>13:10:58 AutoIt3Wrapper Finished.
>Exit code: 0    Time: 1.146
```

Thus, my interpretation is that ContinueCase causes execution to "fall through" and execute the immediate next case section. It doesn't evaluate the Case statement any further.

Reply:
> It doesn't evaluate the Case statement any further.

https://www.autoitscript.com/autoit3/docs/keywords/ContinueCase.htm
`AutoIt does not evaluate the next Case comparison statement - only the code inside the block is run.`

Reply:
Fixed

Reply:
> I even add new `Switch[2].au3` example file to HelpFile.

removed